app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/email/SubscriptionChangeEmailController.scala

```diff
@@ -53,29 +56,30 @@ class SubscriptionChangeEmailController @Inject()(
     with WithAuthAndSessionDataAction
     with Logging
     with SessionUpdates
-    with EmailController[SubscriptionReady, SubscriptionReady] {
+    with EmailController[Subscribed, Subscribed] {
 
   override val isAmendJourney: Boolean = true
 
-  override def validJourney(request: RequestWithSessionData[_]): Either[Result, (SessionData, SubscriptionReady)] =
+  override def validJourney(request: RequestWithSessionData[_]): Either[Result, (SessionData, Subscribed)] =
     request.sessionData.flatMap(s => s.journeyStatus.map(s -> _)) match {
-      case Some((sessionData, s: SubscriptionReady)) => Right(sessionData -> s)
-      case _                                         => Left(Redirect(controllers.routes.StartController.start()))
+      case Some((sessionData, s: Subscribed)) => Right(sessionData -> s)
+      case _                                  => Left(Redirect(controllers.routes.StartController.start()))
     }
 
   override def validVerificationCompleteJourney(
     request: RequestWithSessionData[_]
-  ): Either[Result, (SessionData, SubscriptionReady)] =
+  ): Either[Result, (SessionData, Subscribed)] =
     validJourney(request)
 
-  val subscriptionReadyEmailLens: Lens[SubscriptionReady, Email] =
-    lens[SubscriptionReady].subscriptionDetails.emailAddress
+  override def updateEmail(journey: Subscribed, email: Email)(
+    implicit hc: HeaderCarrier
+  ): EitherT[Future, Error, Subscribed] =
+    subscriptionService
+      .updateSubscribedDetails(journey.subscribedDetails)
```

Comment:
wouldn't you need to do something like `journey.subscribedDetails.copy(email = email)` here?

Reply:
but also, this is the wrong controller to do this in. We need a new controller for changing emails when subscribed - this one if for the subscription journey. We need something like `SubscribedChangeEmailController` which has the backlinks and continueUrl pointing to the homepage. Then we need routes for those endpionts in `app.routes` and the homepage template then needs the change email link stuck in there

Reply:
OK, thanks for the feedback. When we discussed this originally we were looking at the SubscriptionChangeEmailController file and I thought I had to change that. But this makes sense I will change it.

Reply:
created the new controller class with the back link and continue link set appropriately